### PR TITLE
Add command to set reminder, and a script to broadcast reminders

### LINKED ISSUE #13: Reminder
Requirement:
* /slash-command
* Default Time/ Allow user to set timezone
* How often: One-off / Interval 

**Message Format:**
 * /remind [timeformat] [message] # simple reminder once off
 * /remindSetTimezone [timezone] (e.g Australia/Brisbane)
 * /remindCheckTimezone
 * /remindList # return list of reminder with reminder id
 * /remindRemove [reminderId]

**TimeFormat:**
* 1m/h/d/w/M (current day + timeinterval)
* dd/MM/YYYY HH:mm

**TimeInterval:**
* 1m/h/d/w/M (repeat every ****)
* can extend to be more advanced if we can find a library that handle this for us

**Timezone:** 
* tz database time zones
* https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

**reminderId**
* CUID 

**message**
* Message can be anything / text / link / message that mentioned s.o 
 
Note: the reminder is set on a specific channel that it is created from (channel specific)

### PR DESCRIPTION
**Pull request by samhwang**
## Description
This PR will add 2 commands and a script to broadcast reminders..

### Commands

- `reminder`: with 4 subcommands:
  - `on`: Save a reminder on a date + time.
  - `in`: Save a reminder on an X duration from the current time.
  - `list`: List the current user's reminder on the current server.
  - `delete`: Delete a reminder by ID. This can only be done by the user that owns the reminder.
- `server-settings`: with 1 subcommand:
  - `reminder-channel`: ADMIN COMMAND. This will set the channel that the reminder messages will be broadcasted into.

### Script

- The script is under `src/broadcast-reminder.ts`
- It will be bundled along with the bot image upon the build & deploy stage.
- To invoke this script locally, run `npx tsx src/broadcast-reminder.ts`.
- To run this on the production docker image, run `docker compose up broadcast-reminder`.
  - I'd recommend to set up a cron tab that runs every minute that runs this script.

### Notes:
- All reminder records saved in our db are server-specific, so we can easily test and modified the test ones without touching the real production database.

## Motivation and Context
This will let the server users be able to register reminders, a feature long waited for for the bot..

## Related Issue
Resolves #13 

## How Has This Been Tested?
Ran unit test in the bot and make sure all tests passed. Also E2E tested in the test server with the commands to save, list and delete reminders, and running the broadcast command manually.

## Screenshots (if appropriate):
Screen Shot 2023-04-11 at 11 57 29 pm

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/commands/reminder/reminder-utils.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  timestamp: number;
+};
+export const saveReminder = async ({
+  userId,
+  guildId,
+  message,
+  timestamp,
+}: SaveReminderInput): OpResult<Reminder> => {
+  try {
+    const currentDate = new Date();
```
Comment (hieu-cognitia) [inline src/commands/reminder/reminder-utils.ts:19]: Need to also convert this to UNIX timestamp before comparision.  Maybe:
`const currentDate = getUnixTime(new Date());`

**Review by hieu-cognitia — CHANGES_REQUESTED** [review]
Code working well, except for that small bug.
Maybe we can change command to:
* `/reminder on` - Set reminder on a specific date.
* `/reminder in` - Set reminder in ... minutes/hours/days.

**Comment by hieu-cognitia** [comment]
Also reminder only work up until 2038. Please fix this bug to further future proofing this project.
https://en.wikipedia.org/wiki/Year_2038_problem

/s

File: src/commands/reminder/reminder-utils.ts (hunk 1 above)
Comment (samhwang) [reply src/commands/reminder/reminder-utils.ts]: Resolved.

**Review by hieu-cognitia — APPROVED** [review]
Looking good now!

**Pull request merged**